### PR TITLE
fix IPv6 binding problem for eventmachine websockets

### PR DESCRIPTION
**Pull request by muhammadn**
Problem is that eventmachine does not accept IPv6 address with square brackets, (eg. [::1]). Only without it (eg ::1)

This PR checks if there is an IPv6 address with square brackets in the binding port flag and removes the square brackets.

### CODE REVIEW

**Comment by monotek** [comment]
As i mentioned earlier in https://github.com/zammad/zammad/issues/1687#issuecomment-352793649 it also does not work without the square brackets.

File: script/websocket-server.rb [hunk 1]
```diff
@@ -120,7 +120,11 @@ def after_fork(dir)
 @clients = {}
 Rails.configuration.interface = 'websocket'
 EventMachine.run do
-  EventMachine::WebSocket.start( host: @options[:b], port: @options[:p], secure: @options[:s], tls_options: tls_options ) do |ws|
+
+  if /(\[(?:[:0-9A-Fa-f]+)\])/.match?(@options[:b]) # check if there is an IPv6 Address
```
Comment (monotek) [inline script/websocket-server.rb]: Checking ipv6 adresses by own regex looks wrong to me.
We could use resolv here: http://ruby-doc.org/stdlib-2.4.1/libdoc/resolv/rdoc/Resolv/IPv6.html
Example: https://stackoverflow.com/questions/3634998/how-do-i-check-whether-a-value-in-a-string-is-an-ip-address/15157862#15157862

**Comment by muhammadn** [comment]
I have updated the code to use the standard built-in ruby class that parses and removes any square brackets and also shortens the code resulting in less errors than a own-rolled regex:

https://docs.ruby-lang.org/en/2.0.0/IPAddr.html

On my Mac:
Invalid IP -
```
─$ bundle exec script/websocket-server.rb -b '[::not:ipv6:addres]' -p 6389 start
bundler: failed to load command: script/websocket-server.rb (script/websocket-server.rb)
IPAddr::InvalidAddressError: invalid address
```

Valid IP:
```
╰─$ bundle exec script/websocket-server.rb -b '[::]' -p 6389 start                                                                1 ↵
2017-12-24T16:29:46Z:client(-) check unused idle connections...
```

When depending on the IPAddr class, it will also check for valid/invalid IPv4 addresses as well.

Have a merry christmas and happy new year. 🎅🎄

**Comment by thorsteneckel** [comment]
@rlue - I think we can close this one now?

**Comment by rlue** [comment]
Yes, the changes presented in this PR were refactored and merged separately in cd4dc31.

However, test coverage is currently excluded from the Travis CI build (see 2662c17), since it currently does not support IPv6.

**Pull request closed without merging**